### ISSUE TITLE
Automatic formatting

### ISSUE THREAD

**Issue by sryze** [square/spacecommander#60]
Space Commander is great, but one downside I found is that I have to manually execute commands from a terminal to fix formatting problems. Even though it tells me which command to run I still need to copy and paste it into the terminal (or type manually).

Is there a less painful way of using Space Commander, e.g. a GUI or Xcode plugin?

Running it as a build step is one option, however, it destroys your undo history inside Xcode: if you hit Build and Xcode reloads your file you can't use undo.

**Comment by alanf** [comment]
Hi @sryze 

The manual step was to err on the side of caution so as to not modify someone's changes unexpectedly. This was important earlier in Space Commander's development when it was still being tested, so as to not leave the user with a "bad" formatting result and no easy way to undo it.

Anyway, that's the background. I like the idea of a plugin, and I've also seen some modify the pre-commit hook to automatically format the code. This goes against the pre-commit guidelines that git provides though (in practicality it's fine).

I'm definitely open to either a plugin, or a command to install SpaceCommander to automatically format as part of the pre-commit hook. Would either work from your point of view, or is one more strongly preferred than the other?

**Comment by sryze** [comment]
Yes, either one would be fine. Though I'm not sure about an Xcode plugin solution, it seems that plugins are very limited in Xcode 8 (i.e. editor extensions).

Maybe it could also alternatively show a prompt like "Do you want to format these files now?" inside pre-commit hook if it detects bad formatting?

**Comment by johndpope** [comment]
@sryze you can try https://github.com/benoitsan/BBUncrustifyPlugin-Xcode

**Comment by whoyoung** [comment]
@sryze i think my repository [whoyoung/spacecommander](https://github.com/whoyoung/spacecommander) forked from square/spacecommander can meet your demand.
